### ISSUE TITLE
Notice: Undefined index: item in /var/www/......./ShoppingList_backend/api.php on line 11

### ISSUE THREAD

**Issue by ecw2-4** [GroundApps/ShoppingList#35]
Fresh install of the app. Fresh git clone. I enter the authentication in the app. Going to .../install.php on the server, I filled in the details (I'm using sqlite). When trying to connect, the app shows me the following error. 

"The response did not make sense "Notice: Undefined index: item in /var/www/............./ShoppingList_backend/api.php on line 11 Notice: Undefined index: count in /var/ww""

The rest of the error is cut off. 

From using Wireshark with the filter "http.method == "POST"" at the same time, I can see that the POST being sent is auth=.... and function=listall. So that seems OK.

How can I help debugging this?

**Comment by ecw2-4** [comment]
I've been searching further. I have been able to get rid of the "Undefined index" error by altering the api.php and wrapping the itemname and itemcount variable in an isset if structure.

```
if(isset($_POST['item'])) {
        $itemName = $_POST['item'];
        }

if(isset($_POST['count'])) {
        $itemCount = $_POST['count'];
        }
```

That removes the two notices about the undefined index, and show me more of the actual output. Now I see in the app:

```
The response did not make sense: []
```

Experimenting further: I made a file called api2.php, where I replaced all the php POST reference with GET. This allowed me to test the api in a regular browser. I succesfully added something to the database by going to the url: http://server/api2.php?auth=123&item=test&count=2&function=save

When I refresh the app now I see

```
The response did not make sense: [{"item":"test","count":2}]
```

So it looks like the the db works, auth is ok. It's just like the app doesn't pick up the response.

Could this perhaps have something to do with my backend setup? I'm running it on Gentoo, PHP Version 5.6.12-pl0-gentoo

**Comment by ghost** [comment]
We have an error in the app.
We will fixed it in one week when we come back from holidays

**Comment by ghost** [comment]
@mfyorp, In the meantime the earlier version offered in F-Droid works just fine for me.

**Comment by ghost** [comment]
We really apologize for that bug.
We will fixed it as soon as possible

**Comment by ghost** [comment]
Hey, shit happens, man. Thanks for all your work on the app.

**Comment by ecw2-4** [comment]
I updated to 1.3. The backend is up to date as well (git pull and fresh install.php). The issue stil remains for me: ""The response did not make sense "Notice: Undefined index: item in /var/www/............./ShoppingList_backend/api.php on line 13 Notice: Undefined index: count in /var/ww"

If I do the trick as described in the 2nd post in this topic, it works. I changed api.php to:

```
if(isset($_POST['item'])) {
        $itemName = $_POST['item'];
        }
if(isset($_POST['count'])) {
        $itemCount = $_POST['count'];
        }
if(isset($_POST['jsonArray'])) {
        $jsonData = $_POST['jsonArray'];
        }
```

I don't know enough about PHP or GIT to actually contribute. I don't really know what I'm doing.  I just know that this fixed the issue for me.
